gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -1033,3 +1030,99 @@ message Shape {
+  optional TranslatedString stop_name = 3;
+  optional TranslatedString tts_stop_name = 4;
+  optional TranslatedString stop_desc = 5;
+  optional float stop_lat = 6;
+  optional float stop_lon = 7;
+  optional string zone_id = 8;
+  optional TranslatedString stop_url = 9;
+  optional string parent_station = 11;
+  optional string stop_timezone = 12;
+  optional WheelchairBoarding wheelchair_boarding = 13 [default = UNKNOWN];
+  optional string level_id = 14;
+  optional TranslatedString platform_code = 15;
+
+  // The extensions namespace allows 3rd-party developers to extend the
+  // GTFS Realtime Specification in order to add and evaluate new features and
+  // modifications to the spec.
+  extensions 1000 to 1999;
+
+  // The following extension IDs are reserved for private use by any organization.
+  extensions 9000 to 9999;
+}
+
+message TripModifications {
+	message Modification {
+    // The first stop sequence of the original trip that is to be affected by this modification.
+		optional uint32 start_stop_sequence = 1;
+
+    // The number of stops which are canceled and replaced by the replacement_stops. May be zero to indicate no skipped stops.
+		optional uint32 num_stops_replaced = 2;
+
+    // The number of seconds of delay to add to all departure and arrival times following the end of this modification. If multiple modifications apply to the same trip, the delays accumulate as the trip advances. 
+		optional int32 propagated_modification_delay = 3 [default = 0];
+
+    // A list of replacement stops, replacing those of the original trip. The length of the new stop times may be less, the same, or greater than the number of replaced stop times. 
+		repeated ReplacementStop replacement_stops = 4;
+
+		// The extensions namespace allows 3rd-party developers to extend the
+		// GTFS Realtime Specification in order to add and evaluate new features and
+		// modifications to the spec.
+		extensions 1000 to 1999;
+
+		// The following extension IDs are reserved for private use by any organization.
+		extensions 9000 to 9999;
+	}
+
+  // An arbitrary string to identify this detour. The modifications_id MUST be unique for all currently active detours. The ID of each replacement trip is generated by concatenating modifications_id + '_' + trip_id.
+  required string modifications_id = 1;
+
+   // A list of trips affected by this detour. All trips linked by those trip_ids MUST have the same stop pattern. Two trips have the same stop pattern, if they visit the same stops in the same order, and have identical stop sequences.
+  repeated string trip_ids = 2;
+
+  // Dates on which the detour occurs, in the YYYYMMDD format. Producers SHOULD only transmit detours occurring within the next week.
+  repeated string service_dates = 3;
+
+  // The ID of the new shape for the modified trips. May refer to a new shape added using a GTFS-RT Shape message, or to an existing shape defined in the GTFS-Static feed’s shapes.txt.
+  optional string shape_id = 4;
```

Comment:
Is this intended to be the shape of just the modified portion of the trip or the entire trip?

Reply:
The shape will be used for the entire trip.